extensions/amp-base-carousel/1.0/base-carousel.js

```diff
@@ -35,6 +36,7 @@ export function BaseCarousel({
   const childrenArray = toChildArray(children);
   const {length} = childrenArray;
   const [curSlide, setCurSlide] = useState(0);
+  const scrollRef = createRef();
```

Comment:
Where is this ref used? And why is it `createRef` and not `useRef`?

Reply:
Same question. I think `useRef` is necessary to avoid new Ref instances from being created (and the necessary cleanup/setup that's required from getting a new instance).

Reply:
The React docs use `createRef` paired with `forwardRef`, but this appears to be a distinction between class components and functional components. Restored `useRef`.

Reply:
Correct: `createRef` hails from the class-component era and is still occasionally useful. But `useRef` is preferred.